crates/driver/src/infra/solver/dto/notification.rs

```diff
@@ -0,0 +1,40 @@
+use {
+    crate::{domain::competition, infra::notify},
+    serde::Serialize,
+    serde_with::serde_as,
+};
+
+impl Notification {
+    pub fn new(auction_id: Option<competition::auction::Id>, kind: notify::Kind) -> Self {
+        Self {
+            auction_id: auction_id.as_ref().map(ToString::to_string),
+            kind: match kind {
+                notify::Kind::EmptySolution => Kind::EmptySolution,
+                notify::Kind::PriceViolation => Kind::PriceViolation,
+                notify::Kind::ScoringFailed => Kind::ScoringFailed,
+                notify::Kind::UntrustedInternalization => Kind::UntrustedInternalization,
+                notify::Kind::InsufficientBalance => Kind::InsufficientBalance,
+            },
+        }
+    }
+}
+
+#[serde_as]
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Notification {
+    auction_id: Option<String>,
```

Comment:
I think we should always have some identifier for the notify call (not optional)

Reply:
Is it for the sake of debugging the driver - solvers communication?
Asking because notifications generated by the driver would have the `ID`, but notifications generated by the `solvers` won't, at least I don't know how to synchronize creating unique `ID` between two servers (from the top of my head).

Reply:
Let's add it later if we figure out we need it.